pkg/brokerapi/openservicebroker/open_service_broker_client.go

```diff
@@ -78,59 +116,78 @@ func (c *openServiceBrokerClient) GetCatalog() (*brokerapi.Catalog, error) {
 
-	glog.Infof("Doing a request to: %s", url)
-	resp, err := c.client.Do(createHTTPReq)
-	if err != nil {
-		return nil, err
+	createServiceInstanceResponse := brokerapi.CreateServiceInstanceResponse{}
+	if err := util.ResponseBodyToObject(resp, &createServiceInstanceResponse); err != nil {
+		return nil, errResponse{message: err.Error()}
 	}
-	defer resp.Body.Close()
 
-	// TODO: Align this with the actual response model.
-	si := brokerapi.ServiceInstance{}
-	if err := util.ResponseBodyToObject(resp, &si); err != nil {
-		return nil, err
+	switch resp.StatusCode {
+	case http.StatusCreated:
+		return &createServiceInstanceResponse, nil
+	case http.StatusOK:
+		return &createServiceInstanceResponse, nil
+	case http.StatusAccepted:
+		glog.V(3).Infof("Asynchronous response received. Polling broker.")
+		if err := c.pollBroker(ID, createServiceInstanceResponse.Operation); err != nil {
+			return nil, err
+		}
+
+		return &createServiceInstanceResponse, nil
+	case http.StatusConflict:
+		return nil, errConflict
+	case http.StatusUnprocessableEntity:
+		return nil, errAsynchronous
+	default:
+		return nil, errStatusCode{statusCode: resp.StatusCode}
 	}
-	return &si, nil
 }
 
-func (c *openServiceBrokerClient) UpdateServiceInstance(ID string, req *brokerapi.ServiceInstanceRequest) (*brokerapi.ServiceInstance, error) {
+func (c *openServiceBrokerClient) UpdateServiceInstance(ID string, req *brokerapi.CreateServiceInstanceRequest) (*brokerapi.ServiceInstance, error) {
 	// TODO: https://github.com/kubernetes-incubator/service-catalog/issues/114
 	return nil, fmt.Errorf("Not implemented")
 }
 
-func (c *openServiceBrokerClient) DeleteServiceInstance(ID string) error {
+func (c *openServiceBrokerClient) DeleteServiceInstance(ID string, req *brokerapi.DeleteServiceInstanceRequest) error {
 	url := fmt.Sprintf(serviceInstanceFormatString, c.broker.Spec.URL, ID)
-
 	// TODO: Handle the auth
-	deleteHTTPReq, err := http.NewRequest("DELETE", url, nil)
+	resp, err := util.SendRequest(c.client, http.MethodDelete, url, req)
 	if err != nil {
-		glog.Errorf("Failed to create new HTTP request: %v", err)
-		return err
+		return errRequest{message: err.Error()}
 	}
+	defer resp.Body.Close()
 
-	glog.Infof("Doing a request to: %s", url)
-	resp, err := c.client.Do(deleteHTTPReq)
-	if err != nil {
-		glog.Errorf("Failed to DELETE: %#v", err)
-		return err
+	deleteServiceInstanceResponse := brokerapi.DeleteServiceInstanceResponse{}
+	if err := util.ResponseBodyToObject(resp, &deleteServiceInstanceResponse); err != nil {
+		return errResponse{message: err.Error()}
 	}
-	defer resp.Body.Close()
 
-	return nil
+	switch resp.StatusCode {
+	case http.StatusOK:
+		return nil
+	case http.StatusAccepted:
+		glog.V(3).Infof("Asynchronous response received. Polling broker.")
+		if err := c.pollBroker(ID, deleteServiceInstanceResponse.Operation); err != nil {
+			return err
+		}
+
+		return nil
+	case http.StatusGone:
+		// TODO: Should this be treated differently than 200 OK?
```

Comment:
I think you can remove this comment, or turn it into a statement instead of a question. I think the API is meant to be idempotent so getting a "Gone" is ok and the same as a 200.